server/routes/prisons/prisons.ts

```diff
@@ -18,14 +19,41 @@ export default function routes({ prisonService }: Services): Router {
 
     const message = req.flash('message')
 
-    res.render('pages/prisons/prisons', { prisons, prisonNames, message })
+    res.render('pages/prisons/prisons', {
+      errors: req.flash('errors'),
+      prisons,
+      prisonNames,
+      message,
+    })
   })
 
   post('/', async (req, res) => {
-    const prisonCode = req.body.prisonId.toUpperCase()
+    await body('prisonId')
+      .trim()
+      .toUpperCase()
+      .matches(/^[A-Z]{3}$/)
+      .withMessage('Enter three letter prison code')
+      .bail()
+      .custom(async prisonId => {
```

Comment:
I'm running this set of validations [manually](https://express-validator.github.io/docs/guides/manually-running) rather than as Express middleware so that I can have access to `res` to use in `prisonService.getPrisonName()`. This call effectively checks whether a prison ID exists in the Prison Register. It can't be added if it doesn't exist - so we don't get unusable prison codes added (that might then go on to cause issues in Staff UI, etc)